--- sensor.orig	2021-05-20 00:33:26.000000000 +0900
+++ sensor.py	2021-06-03 01:51:58.712402339 +0900
@@ -4,6 +4,7 @@
 import email
 import imaplib
 import logging
+import base64
 
 import voluptuous as vol
 
@@ -227,7 +228,12 @@
                 message_untyped_text = part.get_payload()
 
         if message_text is not None:
-            return message_text
+            if email_message["Content-Transfer-Encoding"] == "base64":
+                _LOGGER.debug("decoded by base64")
+                return base64.b64decode(message_text).decode('utf-8')
+            else:
+                _LOGGER.debug("not decoded")
+                return message_text
 
         if message_html is not None:
             return message_html
