--- sensor.orig	2021-05-20 00:33:26.000000000 +0900
+++ sensor.py	2021-06-03 01:51:58.712402339 +0900
@@ -4,6 +4,7 @@
 import email
 import imaplib
 import logging
+import base64
 
 import voluptuous as vol
 
@@ -227,7 +228,12 @@
                 message_untyped_text = part.get_payload()
 
         if message_text is not None:
-            return message_text
+            if email_message["Content-Transfer-Encoding"] == "base64":
+                _LOGGER.debug("decoded by base64")
+                return base64.b64decode(message_text).decode('utf-8')
+            else:
+                _LOGGER.debug("not decoded")
+                return message_text
 
         if message_html is not None:
             return message_html
@@ -241,12 +247,16 @@
         """Read emails and publish state change."""
         email_message = self._email_reader.read_next()
 
+        _LOGGER.debug("email_message: <%s>", email_message)
+
         if email_message is None:
             self._message = None
             self._state_attributes = {}
             return
 
         if self.sender_allowed(email_message):
+            _LOGGER.debug("allowed the sender")
+
             message = EmailContentSensor.get_msg_subject(email_message)
 
             if self._value_template is not None:
@@ -259,3 +269,5 @@
                 ATTR_DATE: email_message["Date"],
                 ATTR_BODY: EmailContentSensor.get_msg_text(email_message),
             }
+        else:
+            _LOGGER.debug("not allowed the sender")
